### PR TITLE
Refactor: VaultSettled & VaultLiquidated events

### PR DESCRIPTION
**Pull request by CruzMolina**
- refactor: index `vaultType` for `VaultSettled` event, scope variables before emitting event to avoid stack too deep error
- docs: update docs for `VaultSettled` change
- refactor: emit `vaultId` for `VaultLiquidated` event, misc style changes
- docs: update docs for `VaultLiquidated` change
- refactor: rm nested scope from `Controller.settleVault`
- refactor: rename `payoutRecipient` -> `to` in `VaultSettled`, `payoutRecipient` -> `receiver` in `VaultLiquidated` events
- docs: update docs

### Code

- [X] Does your code follow the naming and code documentation guidelines?

### Documentation

- [X] Is your code up to date with the spec?

### CODE REVIEW

**Review by antoncoding — APPROVED** [review]

**Comment by haythem96** [comment]
let's wait on merging this until we see that we need it.

**Comment by CruzMolina** [comment]
@haythem96 While we no longer need these changes for the v0.1 liq bot, these changes are still helpful for UIs and other developers (same reasoning as before).

**Comment by antoncoding** [comment]
we will need vault id for liquidation event

File: contracts/core/Controller.sol [hunk 1]
```diff
@@ -169,21 +169,23 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
     );
     /// @notice emits an event when a vault is settled
     event VaultSettled(
-        address indexed AccountOwner,
-        address indexed to,
-        address indexed otoken,
+        address indexed accountOwner,
+        address indexed oTokenAddress,
+        address payoutRecipient,
```
Comment (haythem96) [inline contracts/core/Controller.sol:174]: Can we keep this to `to` to match the action argument ? IMO better to keep the same naming whenever we can.

File: contracts/core/Controller.sol [hunk 2]
```diff
@@ -169,21 +169,23 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
...
+        uint256 payout,
         uint256 vaultId,
-        uint256 payout
+        uint256 indexed vaultType
     );
     /// @notice emits an event when a vault is liquidated
     event VaultLiquidated(
         address indexed liquidator,
-        address indexed receiver,
+        address indexed payoutRecipient,
```
Comment (haythem96) [inline contracts/core/Controller.sol:182]: Same here, can we keep it receiver to match the same name used in the action argument.

File: contracts/core/Controller.sol [hunk 3]
```diff
@@ -899,7 +901,14 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
 
         pool.transferToUser(collateral, _args.to, payout);
 
-        emit VaultSettled(_args.owner, _args.to, address(otoken), _args.vaultId, payout);
+        uint256 vaultId;
+        address payoutRecipient;
+        {
+            vaultId = _args.vaultId;
+            payoutRecipient = _args.to;
+        }
```
Comment (haythem96) [inline contracts/core/Controller.sol:909]: I think there is no need for this nested scope, it should work fine by declaring the variable directly outside.

**Review by haythem96 — CHANGES_REQUESTED** [review]
Good work, just left few comments.

**Review by antoncoding — APPROVED** [review]
Agree with Haythem's comments

File: contracts/core/Controller.sol (hunk 2 above)
Comment (CruzMolina) [reply contracts/core/Controller.sol:182]: Are you open to renaming it to `payoutRecipient`? It's more explicit.

File: contracts/core/Controller.sol (hunk 1 above)
Comment (CruzMolina) [reply contracts/core/Controller.sol:174]: Same renaming suggestion here. `payoutRecipient` is much more explicit than `to` :joy:.

Comment (haythem96) [reply contracts/core/Controller.sol:174]: I agree, but I guess it require some changes in Controller that is not worth it IMO.

File: contracts/core/Controller.sol [hunk 4]
```diff
@@ -169,21 +169,23 @@ contract Controller is Initializable, OwnableUpgradeSafe, ReentrancyGuardUpgrade
...
+        uint256 payout,
         uint256 vaultId,
-        uint256 payout
+        uint256 indexed vaultType
     );
     /// @notice emits an event when a vault is liquidated
     event VaultLiquidated(
         address indexed liquidator,
-        address indexed receiver,
+        address indexed to,
```
Comment (haythem96) [inline contracts/core/Controller.sol:182]: This called `receiver` in the Action lib

**Review by haythem96 — APPROVED** [review]
LGTM! just left one other comment.
For docs, in the future, will be great to just do it in a separate PR.

File: contracts/core/Controller.sol (hunk 4 above)
Comment (CruzMolina) [reply contracts/core/Controller.sol]: Thanks for catching that!

**Pull request merged**